base/linalg/dense.jl

```diff
@@ -833,9 +833,7 @@ function cosh(A::AbstractMatrix)
     if ishermitian(A)
         return full(cosh(Hermitian(A)))
     end
-    X = exp(A)
-    X .= (X .+ inv(X)) ./ 2
-    return X
+    return (exp(A) .+ exp!(-A)) ./ 2
```

Comment:
You should still do `X = exp(A)` and `X .= ...`, to avoid allocating an extra matrix for the result.

Reply:
Right, thanks!